src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1170,43 +1169,46 @@ public bool DoIndexingWork(IndexingStatsScope stats, CancellationToken cancellat
                                         _mre.Set();
                                 }
                             }
-                        }
-                        finally
-                        {
+
                             if (writeOperation.IsValueCreated)
                             {
                                 using (var indexWriteOperation = writeOperation.Value)
                                 {
                                     indexWriteOperation.Commit(stats);
                                 }
                             }
-                        }
-
-                        _indexStorage.WriteReferences(CurrentIndexingScope.Current, tx);
-                    }
 
-                    using (stats.For(IndexingOperation.Storage.Commit))
-                    {
-                        tx.InnerTransaction.LowLevelTransaction.RetrieveCommitStats(out CommitStats commitStats);
+                            _indexStorage.WriteReferences(CurrentIndexingScope.Current, tx);
+                        }
 
-                        tx.InnerTransaction.LowLevelTransaction.AfterCommitWhenNewReadTransactionsPrevented += () =>
+                        using (stats.For(IndexingOperation.Storage.Commit))
                         {
-                            if (writeOperation.IsValueCreated)
+                            tx.InnerTransaction.LowLevelTransaction.RetrieveCommitStats(out CommitStats commitStats);
+
+                            tx.InnerTransaction.LowLevelTransaction.AfterCommitWhenNewReadTransactionsPrevented += () =>
```

Comment:
Hmm.. I just moved this part of code inside `try catch`. I haven't changed the logic regarding `RecreateSearcher` call. 

Note the comment below "we need to recreate it after transaction....". I don't see how it could be called differently.